fastapi/routing.py

```diff
@@ -1204,6 +1204,7 @@ def read_users():
         app.include_router(internal_router)
         ```
         """
+        assert id(self) != id(router), "Router cannot be the same as parent"
```

Comment:
would `self is not router` not be sufficient?

Reply:
That's completely true, updated!